includes/filesystem.inc

```diff
@@ -264,9 +264,10 @@ function _drush_recursive_copy($src, $dest) {
   elseif (is_link($src)) {
     symlink(readlink($src), $dest);
   }
-  elseif (!(copy($src, $dest) && touch($dest, filemtime($src)))) {
+  elseif (!copy($src, $dest)) {
     return FALSE;
   }
+  touch($dest, filemtime($src));
```

Comment:
Lets add a comment about why we do it this way. Can be a link to #1146. Otherwise, this looks good. Thanks for the explanation.